mixer/cmd/build.go

```diff
@@ -110,6 +112,144 @@ var buildBundlesCmd = &cobra.Command{
+			fail(err)
+		}
+
+		fmt.Println(" Backing up full groups.ini")
+		// Back up groups.ini in case we have deprecated bundles to delete
+		if err = builder.CopyFullGroupsINI(b); err != nil {
+			fail(err)
+		}
+		// Fill this in w/Update bundle definitions
+		// if err := UpdateBudlesForFormatBump(); err != nil {...}
+
+		// Build the +20 (first build in new format) bundles
+		if err = buildBundles(b, buildFlags.noSigning); err != nil {
+			fail(err)
+		}
+
+		fmt.Println("Backing up trimmed groups.ini for future build")
+		// Copy the new trimmed groups.ini to use for the +20 build which will
+		// not have manifest created for fully deleted bundles
+		if err = builder.CopyTrimmedGroupsINI(b); err != nil {
+			fail(err)
+		}
+
+		// Copy +20 chroots to +10 so we can build last formatN build with the
+		// same content
+		prevVersion, err := strconv.Atoi(b.MixVer)
+		if err != nil {
+			fail(err)
+		}
+		prevVersion -= 10
+		source := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer)
+		dest := filepath.Join(b.Config.Builder.ServerStateDir, "image", strconv.Itoa(prevVersion))
+		fmt.Println(" Copying +20 chroots to +10 chroots")
+		if err = helpers.RunCommandSilent("cp", "-al", source, dest); err != nil {
+			failf("Failed to copy +20 chroots to +10: %s\n", err)
+		}
+
+		// Copy the old groups.ini file back which contains ALL original bundle names
+		// to account for any removed bundles in this build when creating manifests
+		fmt.Println(" Copying full groups.ini back to working directory")
+		if err = builder.RevertFullGroupsINI(b); err != nil {
+			fail(err)
+		}
+
+		// Set the format back to the previous format version before building the +10 update
+		prevFormat, err := strconv.Atoi(buildFlags.format)
+		if err != nil {
+			fail(err)
+		}
+		prevFormat--
+		if err = builder.UpdateFormatVersion(b, strconv.Itoa(prevFormat)); err != nil {
+			fail(err)
+		}
+		// Set mixversion to the +10 since we have used +20 up to this point
+		if err = b.DecrementMixVer(); err != nil {
+			fail(err)
+		}
+		// Re-read builder.conf after updating format/mixver
+		b, err = builder.NewFromConfig(config)
+		if err != nil {
+			fail(err)
+		}
+		setWorkers(b)
+
+		// Build the update content for the +10 build
+		err = b.BuildUpdate(buildFlags.prefix, buildFlags.minVersion, buildFlags.format, buildFlags.noSigning, !buildFlags.noPublish, buildFlags.keepChroot)
+		if err != nil {
+			failf("Couldn't build update: %s", err)
+		}
+		// Copy the trimmed groups.ini back for the +20 build to use
+		if err = builder.RevertTrimmedGroupsINI(b); err != nil {
+			fail(err)
+		}
+		// Update the format back to the formatN+1 version for +20 to use
+		if err = builder.UpdateFormatVersion(b, buildFlags.format); err != nil {
+			fail(err)
+		}
+		// Incremement regardless of flag because +20 must be the next version
+		if err = b.UpdateMixVer(); err != nil {
+			failf("Couldn't update Mix Version")
+		}
+	},
+}
+
+var buildNewFormatCmd = &cobra.Command{
```

Comment:
What happens if you call this without having called `mixer build format-old` first?

Reply:
Bad stuff, you don't call this without calling the other :) just as you don't call build update before you call build bundles.

Reply:
We could add a check for the `upstreamversion.bump` file that we discussed and error out early if this doesn't exist.